Source/JavaScriptCore/yarr/YarrParser.h

```diff
@@ -595,12 +722,15 @@ class Parser {
     {
     }
 
-    // The handling of IdentityEscapes is different depending on the unicode flag.
+    // The handling of IdentityEscapes is different depending on which unicode flag if any is active.
     // For Unicode patterns, IdentityEscapes only include SyntaxCharacters or '/'.
+    // For UnicodeSet patterns, adds to the UnicodePatterns IdentityEscapes
+    // ClassSetReservedPunctionation which includes &-!#%,:;<=>@`~
     // For non-unicode patterns, most any character can be escaped.
     bool isIdentityEscapeAnError(int ch)
     {
-        if (isEitherUnicodeCompilation() && (!strchr("^$\\.*+?()[]{}|/", ch) || !ch)) {
+        if (isEitherUnicodeCompilation()
+            && (!strchr(isUnicodeCompilation() ? "^$\\.*+?()[]{}|/" : "^$\\.*+?()[]{}|/&-!#%,:;<=>@`~" , ch) || !ch)) {
```

Comment:
Also, is it correct for non ASCII ch?

Reply:
Part of that is existing code.  But to be safe, I'll add an isASCII() check.